src/state/opamFormatUpgrade.ml

```diff
@@ -1066,22 +1066,33 @@ let downgrade_2_1_switches root conf =
     (OpamFile.Config.installed_switches conf);
   conf
 
-let from_2_1_alpha_to_2_1_alpha2 root conf =
-  downgrade_2_1_switches root conf
+let from_2_1_alpha_to_2_1_alpha2 root ~on_the_fly:_ conf =
+  downgrade_2_1_switches root conf, Some `Switch
+
+let from_2_1_alpha2_to_2_1_rc root ~on_the_fly:_ conf =
+  downgrade_2_1_switches root conf, Some `Switch
+
+let from_2_1_rc_to_2_1 _ ~on_the_fly:_ conf = conf, None
+
+let from_2_0_to_2_1 _ ~on_the_fly:_ conf = conf, None
 
-let from_2_1_alpha2_to_v2_1_rc root conf =
-  downgrade_2_1_switches root conf
+let v2_2_alpha = OpamVersion.of_string "2.2~alpha"
 
-let from_2_1_rc_to_v2_1 _ conf = conf
+let from_2_1_to_2_2_alpha_repo root _conf =
+  let f = OpamPath.repos_config root in
+  OpamStd.Option.iter (OpamFile.Repos_config.write f)
+    (OpamFile.Repos_config.read_opt f)
```

Comment:
Shouldn't this be iterated over to set all of the initialised fields to true, given that they must have been initialised before? (cf. the v1.3+ upgrade)?